anatomy_of_variables/Anatomy of Variables.md

```diff
@@ -0,0 +1,229 @@
+
+Instance Variables are global variables that are **ONLY** used in **Instance Methods** within a class. Instance variables in ruby start with a single  `@` sign. Any variable assigned starting with a single `@` sign can be manipulated within an instance method. Just in case, you don't know what an instance method is in ruby, let's take a quick look.
+
+Ruby instance methods are defined with the keyword `def`, then followed by the name of the method and ended with the keyword `end`. Basically, the common methods in ruby are instance methods. These methods can only be called on an instance of the class they belong to.
+The example below shows an instance variable been used in an instance method.
+
+```ruby
+class User
+    def initialize
+        @name = "Adebayo"
+    end
+
+    def greet
+        puts "Hello #{@name}"
+    end
+end
+
+User.new.greet #=> Hello Adebayo
+```
+
+The above example has instance variable `@name` which holds a name `Adebayo`. The instance method `greet` adds `Hello` to the value of `@name`.  This tells us that instance variables act as global variables **within** a class.  
+
+**Class Variables**
+
+Class variables work exactly like instance methods. These varibales are denoted with a double `@` sign, i.e `@@`. These variables are  **ONLY** used in **Class Methods** within a class. Just in case, you don't know what a class method is in ruby, let's take a quick look.
+
+The keyword `self` makes a class method quite diferent from an instance method. A class method is also defined by the `def` keyword, then the keyword `self` is prpended to the method name. For example, if we were to rewrite the above example as a class method, the method `greet` will be written as `self.greet`. Class methods, however, can be called directly on the class they belong to. The example below shows a class variable been used in a class method.
+
+```ruby
+Class User
+    @@name = Adebayo
+
+    def self.name
+         puts "Hello #{@@name}"
+    end
+end
+
+User.name  #=> "Hello Adebayo"
+```
+
+**Class Instance Variables**
+
+These variables are a combination of the two we have discussed earlier. They are used as global variables but they are assigned as local variables. This means they are assigned in a class method like instance variables but they can be used in any class methods within the same class.
+
+The example below shows how class instance variables are used.
+
+```ruby
+class Person
+    def self.username(name)
+      @name = name
+    end
+
+    def self.greet
+      "Hello #{@name}"
+    end
+end
+
+Person.username("Adebayo") #=> "Adebayo"
+Person.greet #=> "Hello Adebayo"
+
+```
+
+In the snippet above, `name` method was used to get the person's name and it was assigned to `@name` variable. This variable, however, can be used in any other class method within that class.
+
+One important thing to know is that this kind of variables will not be available in classes that inherit from the class in which they are assigned.
```

Comment:
This your examples doesn't prove that they are not available.
Just like you need to call `username` method on Person class before invoking `greet` method, you have to call `username` method in Property class before you invoke `owner` method.

A good example will be

```ruby
 class Person
   @name = "Adebayo"
   def self.greet
       "Hello #{@name}"
    end
 end

Person.greet #=> "Hello Adebayo"

class Student < Person; end
Student.greet #=> "Hello "

class Student < Person
  @name = "Penmuse"
end

Student.greet #=> "Hello Penmuse"
```